### ISSUE TITLE
[Change Request]: HAL's Store script change

### ISSUE THREAD

**Issue by FuckingStealingNames** [igorkis-scrts/A3-Antistasi-Plus#141]
### Is the requested change related to a problem?

With the current code, if going above the limit whilst trying to sell/buy, it will just grey out the button. Can be very annoying to the user.

### Describe the solution you'd like.

Changing the code in
https://github.com/igorkis-scrts/A3-Antistasi-Plus/blob/unstable/A3A/addons/hals/Addons/store/functions/client/ui/fn_main.sqf#L383
up to line 385 with

```
private _ctrlInput = CTRLT(IDC_EDIT);
_amount = _ctrlInput getVariable ["amt", 1];
if (_stock < 1 ||  _amount < 1 || _amount > _stock) exitWith {
  _ctrlInput setVariable ["amt", _stock];
  _ctrlInput ctrlSetText str _stock;
  ["text", ["update", ["buy", [CTRL(IDC_LISTBOX) getVariable "value", CTRLT(IDC_EDIT) getVariable "amt"]]]] call HALs_store_fnc_main;
};
```
will allow an automatic recount to the maximum amount in the list.
Script tested in LAN, and a dedicated server.

### Additional context

What this does is instead of greying out the sell/buy button if you incorrectly write the amount, it will correct itself to the valid maximum (doesn't affect you writing an invalid character to default back to 1).

Current bug with this is the weight line will not update correctly if you don't write the correct amount while attempting to buy.

**Comment by FuckingStealingNames** [comment]
Edited since the coder @BytewaveMLP found out a fix to the previous bug with amount not updating.

He's also looking into the weight line.

**Comment by igorkis-scrts** [comment]
Well, it's a great addition and seems works just fine, tweak will be released in 2.1.10. 
About HAL's store in general - it's a third party store framework and I have really limited knowledge about it's codebase and how it's works, so can't say anything concrete about it.
